src/js/core/player.js

```diff
@@ -792,17 +792,77 @@ function name(): {country: string, firstName: string, lastName: string} {
     const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
     const firstNameRow = playerNames.first[country].find(row => row[1] >= fnRand);
     if (firstNameRow === undefined) {
-        throw new Error(`Undefined firstNameRow (fnRand=${fnRand}`);
+        throw new Error(`Undefined firstNameRow (fnRand=${fnRand})`);
+    }
+    let firstName = firstNameRow[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 25%% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 25%% will use it fully but as a 'nickname'
+        //   e.g. {Xavier 'Yanick' Zed}
```

Comment:
Because it was cheeky and I didn't think you would actually like it.

It's my tongue-in-cheek response to the "why is the first name from a different language than the middle name" problem. E.g. `Jasko "Kenny" Nnamake` which is a legit legally-generated Swedish name according to our data/names.js file, with a Swedish first name, an English middle name, and a Nigerian last name (but all parts belonging to at least one Swedish-born player, such as Olouma Nnamaka).

So wrapping the name in quotes is kind of a "variant" form so that not everything gets automatically turned into a single abbreviation when I can't deal with ambiguity/complexity. My hope is that we might be able to use that to allow some U.S. players to keep their middle names. 

If you actually like it, we could do a scheme like:

* 25% of all middle names are abbreviated to the first initial (`Y.`)
* 50% of all middle names are wrapped in double-quotes (`"Yanick"`)
* 25% of all middle names are kept as-is (`Yanick`)

This way, when we get unfortunately drawn names in the U.S., like Billy Mohammed Jones, we can get variations like Billy M. Jones, Billy "Mohammed" Jones, and Billy Mohammed Jones. Since the *unaltered* variant appears less often, it will be less noticeable to users that sometimes we randomly draw two obviously different names.

If you're OK with that, I would much rather make the double-quotes more common than they are currently.